tsl/src/compression/compression.c

```diff
@@ -2421,40 +2421,55 @@ static bool
 decompress_chunk_walker(PlanState *ps, List *relids)
 {
 	RangeTblEntry *rte = NULL;
+	bool needs_decompression = false;
+	List *predicates = NIL;
 	Chunk *current_chunk;
 	if (ps == NULL)
 		return false;
 
 	switch (nodeTag(ps))
 	{
+		/* Note: IndexOnlyScans will never be selected for target
+		 * tables because system columns are necessary in order to modify the
+		 * data and those columns cannot be a part of the index
+		 */
+		case T_IndexScanState:
+		{
+			predicates = ((IndexScan *) ps->plan)->indexqualorig;
```

Comment:
Does indexqualorig include both index conditions and index filters?

Reply:
It does not, added filter quals to predicates as well.